owncloud/test/test.py

```diff
@@ -429,6 +429,124 @@ def test_move_to_non_existing_dir(self):
             'x'
         )
 
+    def test_copy_in_place(self):
+        """Test copy in place"""
+
+        self.assertTrue(
+            self.client.put_file_contents(
+                self.test_root + 'copy this file!.txt',
+                'to copy'
+            )
+        )
+        self.assertTrue(
+            self.client.copy(
+                self.test_root + 'copy this file!.txt',
+                self.test_root + 'copied in place.txt'
+            )
+        )
+        self.assertEquals(
+            self.client.get_file_contents(
+                self.test_root + 'copied in place.txt'
+            ),
+            'to copy'
+        )
+
+        self.assertEquals(
+            self.client.get_file_contents(
+                self.test_root + 'copy this file!.txt'
+            ),
+            'to copy'
+        )
+
+
+    def test_copy_into_subdir(self):
+        """Test copy into subdir"""
+
+        self.assertTrue(
+            self.client.put_file_contents(
+                self.test_root + 'copy into subdir.txt',
+                'first file'
+            )
+        )
+        self.assertTrue(
+            self.client.mkdir(
+                self.test_root + 'subdir'
+            )
+        )
+        self.assertTrue(
+            self.client.copy(
+                self.test_root + 'copy into subdir.txt',
+                self.test_root + 'subdir/file copied.txt'
+            )
+        )
+        self.assertEquals(
+            self.client.get_file_contents(
+                self.test_root + 'subdir/file copied.txt'
+            ),
+            'first file'
+        )
+        self.assertEquals(
+            self.client.get_file_contents(
+                self.test_root + 'copy into subdir.txt'
+            ),
+            'first file'
+        )
+        
+    def test_copy_unicode(self):
+        """Test copy unicode to dir"""
+        self.assertTrue(
+            self.client.put_file_contents(
+                self.test_root + u'दिलै होस छोरा होस.txt',
+                'content'
+            )
+        )
+        self.assertTrue(
+            self.client.mkdir(
+                self.test_root + 'subdir'
+            )
+        )
+        self.assertTrue(
+            self.client.copy(
+                self.test_root + u'दिलै होस छोरा होस.txt',
```

Comment:
It's a nepali saying.
दिलै होस छोरा होस - You need time for doing something in a good way. - Haste makes waste.